src/services/log.service.ts

```diff
@@ -36,21 +36,27 @@ export class LogService {
      * @param createLogDto class that carries the request body for a Log.
      */
     async create(createLogDto: CreateLogDto): Promise<Log> {
-        const LogEntity = plainToClass(Log, createLogDto);
-        LogEntity.creationTime = new Date();
-        if (LogEntity.attachments) {
-            for (const attachment of LogEntity.attachments) {
-                attachment.creationTime = LogEntity.creationTime;
+        const logEntity = plainToClass(Log, createLogDto);
+        logEntity.creationTime = new Date();
+        logEntity.runs = [];
+        if (logEntity.attachments) {
+            for (const attachment of logEntity.attachments) {
+                attachment.creationTime = logEntity.creationTime;
             }
         }
-        await this.repository.save(LogEntity);
-        return LogEntity;
+        if (createLogDto.runs) {
+            for (const runNumber of createLogDto.runs) {
+                const run = await this.runRepository.findOne(runNumber);
+                await logEntity.runs.push(run);
+            }
+        }
+        return await this.repository.save(logEntity);
     }
 
     /**
      * Returns all Logs from the db.
      */
-    async findAll(queryLogDto: QueryLogDto): Promise<{ logs: Log[], count: number }> {
+    async findAll(queryLogDto: QueryLogDto): Promise<{ logs: Log[], additionalInformation: any }> {
```

Comment:
Replace the `any` for a proper interface that can be reused and represents pagination data to send along with a ResponseObject holding items.